test/distributed/org/apache/cassandra/distributed/impl/MessageFilters.java

```diff
@@ -18,65 +18,48 @@
 
 package org.apache.cassandra.distributed.impl;
 
+import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Set;
-import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
 
-import org.apache.cassandra.distributed.api.IInstance;
+import org.apache.cassandra.distributed.api.IMatcher;
+import org.apache.cassandra.distributed.api.IMessage;
 import org.apache.cassandra.distributed.api.IMessageFilters;
 import org.apache.cassandra.net.Verb;
 
-public class MessageFilters implements IMessageFilters
+public class MessageFilters implements IMessageFilters, IMatcher
 {
-    private final Set<Filter> filters = new CopyOnWriteArraySet<>();
-
-    public boolean permit(IInstance from, IInstance to, int verb)
+    private final IMatcher MATCH_ALL = (from, to, msg) -> true;
+    private final List<Filter> filters = new CopyOnWriteArrayList<>();
+
+    /**
+     * {@code true} value returned by this method implies that the message was not matched by any
```

Comment:
you are documenting the implementation where as it should be the interface; that is what should say the expected semantics.